analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -155,6 +156,11 @@ public override void VisitAttribute(AttributeSyntax node)
                 {
                     TypesUsedWithReflection.Add(typeSymbol);
                 }
+                else if (symbol.ContainingType.GetEffectiveAccessibility() == Accessibility.Private
+                    && symbol.ContainingAssembly.Equals(compilation.Assembly))
```

Comment:
I don't think this check is ever false. It can be removed.

Reply:
It checks for two conditions, which are often false:
- the attribute is a private class
- the attribute is declared in the assembly (e.g. to ignore `System.Diagnostics.DebuggerDisplayAttribute`)

Reply:
That is what I was saying: `symbol.ContainingAssembly.Equals(compilation.Assembly)` is never false because a private class can never be in another assembly.

Reply:
Thanks. I missed that one.